.github/workflows/build-docker-images.yml

```diff
@@ -0,0 +1,67 @@
+name: Build and Push Docker Images
+
+on:
+  push:
+    branches: [develop, 'GEN*', 'gen*']
+    paths:
+      - 'scripts/references/**'
+      - '.github/workflows/build-docker-images.yml'
+  workflow_dispatch:
+
+jobs:
+  build_references_docker:
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: sage-bionetworks/genie-bpc-pipeline
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+        
+    - name: Setup Docker buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Fetch the default branch (develop) for comparison
+      run: git fetch origin develop:refs/remotes/origin/develop --depth=1
+
+    - name: Check for Changes in scripts/references
+      id: check_changes
+      run: |
+        # Check for a merge base, fallback to root commit if none exists
+        if git merge-base --is-ancestor origin/develop HEAD; then
+          DIFF_BASE="origin/develop"
+        else
+          DIFF_BASE=$(git rev-list --max-parents=0 HEAD)  # Use the initial commit as fallback
+        fi
+
+        # Compare changes between DIFF_BASE and HEAD
```

Comment:
I think the comparison results will come into play when we add in other docker images because we don't want to re-build docker images for all when we just modify the code for one of them. Right now I only have one for `scripts/references` but it did work when I pushed edits to this GH actions workflow file and it skipped the build for the docker image.

Reply:
why don't remove path '.github/workflows/build-docker-images.yml'?

Reply:
Mainly keeping it because maybe we'll want the workflow to run even when we don't have changes to docker